sdks/python/apache_beam/io/gcp/bigquery_schema_tools_test.py

```diff
@@ -61,6 +61,25 @@ def test_check_conversion_with_empty_schema(self):
         the_table_schema=schema)
     self.assertEqual(usertype.__annotations__, {})
 
+  def test_check_schema_conversions_with_timestamp(self):
+    fields = [
+        bigquery.TableFieldSchema(name='stn', type='STRING', mode="NULLABLE"),
+        bigquery.TableFieldSchema(name='temp', type='FLOAT64', mode="REPEATED"),
+        bigquery.TableFieldSchema(
+            name='times', type='TIMESTAMP', mode="NULLABLE")
+    ]
+    schema = bigquery.TableSchema(fields=fields)
+
+    usertype = bigquery_schema_tools.generate_user_type_from_bq_schema(
+        the_table_schema=schema)
+    self.assertEqual(
+        usertype.__annotations__,
+        {
+            'stn': typing.Optional[str],
+            'temp': typing.Sequence[np.float64],
+            'times': typing.Any
```

Comment:
I think because this is using the `.from_utc_datetime` method, and datetime isn't defined here https://github.com/apache/beam/blob/master/sdks/python/apache_beam/typehints/schemas.py#L23, the type is shunted to `typing.Any`

Reply:
Ahh I see it changed in [this commit](https://github.com/apache/beam/pull/23070/commits/6881243eec5b307dbb73b0905114547f463b1506#diff-a2c4c0db5641c542c2ce62959c0cca307d810109c982fb1065e123ec8426f296R45). [`from_utc_datetime()`](https://github.com/apache/beam/blob/8a1aad6824319abf662570de36e2580db8e9450e/sdks/python/apache_beam/utils/timestamp.py#L98) returns a `Timestamp` object. Would testing for Optional[Timestamp] work? As it is now the test for timestamp is not very accurate.

Reply:
Hm so when I look at `dict_of_tuples`, I see `[('stn', typing.Optional[str]), ('temp', typing.Sequence[numpy.float64]), ('times', typing.Optional[<bound method Timestamp.from_utc_datetime of <class 'apache_beam.utils.timestamp.Timestamp'>>])]` which essentially becomes `Typing.Any` when calling `named_fields_to_schema` (https://github.com/apache/beam/blob/master/sdks/python/apache_beam/typehints/schemas.py#L120) 


IIUC this is intended

Reply:
@tvalentyn , @TheNeuralBit for any thoughts

Reply:
` typing.Optional[<bound method Timestamp.from_utc_datetime of <class 'apache_beam.utils.timestamp.Timestamp'>>] `
This doesn't sound correct. Why are we not using  apache_beam.utils.timestamp.Timestamp ?

Reply:
It looks like somewhere we're creating a type `Optional[Timestamp.from_utc_datetime]`, which is not a real typehint, as Valentyn indicated it should be `Optional[Timestamp]`